provider/maas/environprovider.go

```diff
@@ -67,6 +68,24 @@ func (p maasEnvironProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+// Ping tests the connection to the cloud, to verify the endpoint is valid.
+func (p maasEnvironProvider) Ping(endpoint string) error {
+
+	c, err := gomaasapi.NewAnonymousClient(endpoint, apiVersion2)
```

Comment:
Note that the maas provider does some 1.0 fallback when it cannot connect to 2.0. This might need to share some of that logic. 

Also keep in mind: lp:1644395 around the way this is exposed to the end user.